### PR TITLE
Refactoring/nico

### PR DESCRIPTION
**Pull request by nicohorn**
Refactored some components and added some documentation. Project is now slightly cleaner.

### CODE REVIEW

File: src/modules/academic-unit/academic-unit-budget-update.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const AcademicUnitBudgetUpdate = ({
             <CurrencyInput
                 defaultPrice={ACBudgets.at(-1)?.amount ?? 0}
                 className="min-w-[7rem] rounded-md py-1 text-xs"
-                priceSetter={(e) => setNewAmount(e)}
+                priceSetter={(price: number) => setNewAmount(price)}
```
[the remark this answers is not in the document]
Comment (bojkomatias) [reply src/modules/academic-unit/academic-unit-budget-update.tsx:86]: priceSetter function already specifies in component (value:number) => void, hence type is already infered

File: src/modules/elements/popover.tsx [hunk 2]
```diff
@@ -20,14 +32,11 @@ export default function PopoverComponent({
...
-                        ? ` ${buttonStyle(
-                              'secondary',
-                              false
-                          )} px-2 py-1 text-xs`
-                        : ''
-                }
+                /*I'm checking the type of the title to be a string because if it isn't (let's say you want to show an icon instead of a title), it should take the classes of the element/component being passed. Otherwise, the title will be styled automatically.*/
+                className={cx(
+                    typeof title == 'string' &&
+                        `${buttonStyle('secondary', false)} px-2 py-1 text-xs`
```
Comment (bojkomatias) [inline src/modules/elements/popover.tsx:38]:
```suggestion
                        buttonStyle('secondary', false), 'px-2 py-1 text-xs'
```

Comment (bojkomatias) [reply src/modules/elements/popover.tsx:38]: buttonStyle returns a string, no need to interpolate, just concatenate

File: src/modules/protocol/elements/inputs/currency-input.tsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ const CurrencyInput = ({
             </div>
             <CurrencyInputElement
                 defaultPrice={form.getInputProps(path).value}
-                priceSetter={(e) => form.setFieldValue(path, e)}
+                priceSetter={(price: number) => form.setFieldValue(path, price)}
```
Comment (bojkomatias) [inline src/modules/protocol/elements/inputs/currency-input.tsx:33]:
```suggestion
                priceSetter={(price) => form.setFieldValue(path, price)}
```

File: src/modules/elements/popover.tsx [hunk 4]
```diff
@@ -41,26 +50,28 @@ export default function PopoverComponent({
...
                                 column
                                     ? 'flex flex-col items-end justify-start gap-2'
                                     : 'flex items-center gap-2'
-                            }`}
+                            )}
                         >
                             {children}
                             <div
+                                /* This is the function that closes the popover, it gets triggered once the action button is clicked. The timeout is for ux purposes */
                                 onClick={async () => {
```
Comment (bojkomatias) [inline src/modules/elements/popover.tsx:71]:
```suggestion
                                onClick={() => {
```

File: src/modules/academic-unit/academic-unit-budget-update.tsx (hunk 1 above)
Comment (bojkomatias) [inline src/modules/academic-unit/academic-unit-budget-update.tsx:86]:
```suggestion
                priceSetter={(price) => setNewAmount(price)}
```

**Review by bojkomatias — CHANGES_REQUESTED** [review]
Good documentation overall

Suggestions remove unnecessary typings, already inferred

File: src/modules/protocol/elements/inputs/currency-input.tsx (hunk 3 above)
Comment (nicohorn) [reply src/modules/protocol/elements/inputs/currency-input.tsx:33]: image

File: src/modules/academic-unit/academic-unit-budget-update.tsx (hunk 1 above)
Comment (nicohorn) [reply src/modules/academic-unit/academic-unit-budget-update.tsx:86]: Same as with my previous comment

File: src/modules/protocol/elements/inputs/currency-input.tsx (hunk 3 above)
Comment (bojkomatias) [reply src/modules/protocol/elements/inputs/currency-input.tsx:33]: Epa epa, que rarooo!

Comment (bojkomatias) [reply src/modules/protocol/elements/inputs/currency-input.tsx:33]: Screen Shot 2023-09-21 at 14 30 33

Tenés esto así? Porque no estaría funcionando la inference

**Review by bojkomatias — APPROVED** [review]
Inference not working on Nico's PC, not a big deal anyways

**Pull request merged**